ssh-agent.c

```diff
@@ -1981,6 +1992,16 @@ cleanup_handler(int sig)
 	_exit(2);
 }
 
+/*ARGSUSED*/
+static void
+remove_all_handler(int sig)
+{
+	if (run_state == RS_POLL)
+		remove_all_identities();
```

Comment:
doing work like this in a signal handler is not safe.  In this case, it should be sufficient to write to the sig_atomic_t, then use ppoll instead of poll to ensure it wakes up on receipt of the signal.